src/Core/src/Stream/RewindableStream.php

```diff
@@ -63,11 +63,26 @@ public function getIterator(): \Traversable
             return;
         }
 
-        $resource = \tmpfile();
+        $resource = \fopen('php://memory', 'r+b');
         $this->fallback = ResourceStream::create($resource);
 
+        $size = 0;
+        $inmemory = true;
         foreach ($this->content as $chunk) {
             \fwrite($resource, $chunk);
+            if ($inmemory) {
+                $size += \strlen($chunk);
+                // switch to filesystem
+                if ($size > 1) {
```

Comment:
```suggestion
                // If string larger that 1 Mb
                if ($this->fallback->length() > 1024*1024) {
```

We also need to close the memory resource

Reply:
haha debug spoted...

I didn't used `$this->fallback->length()` because it's more expensive that a simple counter